### PR TITLE
En 4606 tx value as string

### PR DESCRIPTION
**Pull request by bogdan-rosianu**

### CODE REVIEW

File: integrationTests/multiShard/smartContract/executingSCCalls_test.go [hunk 1]
```diff
@@ -434,7 +435,14 @@ func processAndTestIntermediateResults(t *testing.T, proposerNodeShardSC *testNo
 		txBytes, _ := proposerNodeShardSC.store.Get(dataRetriever.UnsignedTransactionUnit, hash)
 		assert.NotNil(t, txBytes)
 
-		tx := &transaction.Transaction{}
+		//TODO should refactor this, maybe remove the whole integrationTests/multiShard/smartContract package as these tests
+		// are included by integrationTests/multiShard/block package bu should modify the minimum tx gasprice != 0
```
Comment (sasurobert) [inline integrationTests/multiShard/smartContract/executingSCCalls_test.go]: "but"

File: data/transaction/transaction.go [hunk 2]
```diff
@@ -102,7 +97,8 @@ func (tx *Transaction) IsInterfaceNil() bool {
 
 // GetValue returns the value of the transaction
 func (tx *Transaction) GetValue() *big.Int {
-	return tx.Value
+	val, _ := big.NewInt(0).SetString(tx.Value, 10)
+	return val
```
Comment (sasurobert) [inline data/transaction/transaction.go]: this is suboptimal. Every time value is called to create a new big.int. also setString consumes processing.

Comment (iulianpascalau) [reply data/transaction/transaction.go]: Totally agree. I'm not happy of this solution, either.

File: integrationTests/multiShard/smartContract/executingSCCalls_test.go (hunk 1 above)
Comment (iulianpascalau) [reply integrationTests/multiShard/smartContract/executingSCCalls_test.go]: Done

File: data/transaction/transaction.go (hunk 2 above)
Comment (iulianpascalau) [reply data/transaction/transaction.go]: Done, changed back so transaction.Value is *big.Int but transaction has custom methods for marshaling and unmarshaling that treats the Value field as string.

File: integrationTests/frontend/wallet/txInterception_test.go [hunk 3]
```diff
@@ -121,16 +101,17 @@ func testInterceptedTxFromFrontendGeneratedParams(
 
 	integrationTests.MintAddress(node.AccntState, sndAddrBytes, valMinting)
 
-	txHexHash, err = node.SendTransaction(&transaction.Transaction{
+	tx := &transaction.Transaction{
```
Comment (SebastianMarian) [inline integrationTests/frontend/wallet/txInterception_test.go]: You can put back: Value:     frontendValue and remove tx.SetValue(frontendValue)

File: data/transaction/transaction.go [hunk 4]
```diff
@@ -154,3 +156,63 @@ func (tx *Transaction) SetRecvAddress(addr []byte) {
...
+		SndAddr   []byte `json:"sender"`
+		GasPrice  uint64 `json:"gasPrice,omitempty"`
+		GasLimit  uint64 `json:"gasLimit,omitempty"`
+		Data      string `json:"data,omitempty"`
+		Signature []byte `json:"signature,omitempty"`
+	}{}
+	if err := json.Unmarshal(dataBuff, &aux); err != nil {
+		return err
+	}
+	tx.Nonce = aux.Nonce
```
Comment (sasurobert) [inline data/transaction/transaction.go]: add an empty line before

**Pull request merged**